### PR TITLE
Chore: Update developer guide to reflect changes to go get

### PR DESCRIPTION
**Pull request by sakjur**
**What this PR does / why we need it**:

Two changes made to the Go ecosystem have made our current recommendation somewhat outdated

1. Go no longer requires the `$GOPATH` structure, so we can check out the Grafana repository anywhere.
2. The Go module proxy isn't compatible with Grafana's versioning and is enabled by default these days, so downloading Grafana with `go get` in recent Go versions isn't necessarily working.

Ref: https://grafana.slack.com/archives/C05675Y4F/p1595927629296300

### CODE REVIEW

File: contribute/developer-guide.md [hunk 1]
```diff
@@ -27,11 +27,14 @@ npm install -g yarn
...
 
-1. Add `export GOPATH=$HOME/go/` to the bottom of your `$HOME/.bash_profile`.
-1. Open a terminal and run `go get github.com/grafana/grafana` in your terminal. This command downloads, and installs Grafana to your `$GOPATH`.
-1. Open `$GOPATH/src/github.com/grafana/grafana` in your favorite code editor.
+1. Open a terminal and run `git clone https://github.com/grafana/grafana.git`. This command downloads Grafana to a new `grafana` directory in your current directory.
+1. Open the `grafana` directory in your favorite code editor.
+
+For alternative ways of cloning the Grafana repository, please refer to [GitHub's cloning a repository](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository) documentation.
+
+We do not recommend that you use Go's `go get` to download Grafana. Recent versions of Go have added behavior which isn't compatible with the way the Grafana repository is structured.
```
Comment (aknuds1) [inline contribute/developer-guide.md:37]: Nit, but I think the following reads better:
```suggestion
We do not recommend using Go's `go get` to download Grafana. Recent versions of Go have added behavior which isn't compatible with the way the Grafana repository is structured.
```

**Review by aknuds1 — APPROVED** [review]
LGTM with a nit. Thanks for fixing this BTW, wasn't aware it was stale!

File: contribute/developer-guide.md [hunk 2]
```diff
@@ -27,11 +27,14 @@ npm install -g yarn
...
 
-1. Add `export GOPATH=$HOME/go/` to the bottom of your `$HOME/.bash_profile`.
-1. Open a terminal and run `go get github.com/grafana/grafana` in your terminal. This command downloads, and installs Grafana to your `$GOPATH`.
-1. Open `$GOPATH/src/github.com/grafana/grafana` in your favorite code editor.
+1. Open a terminal and run `git clone https://github.com/grafana/grafana.git`. This command downloads Grafana to a new `grafana` directory in your current directory.
+1. Open the `grafana` directory in your favorite code editor.
+
+For alternative ways of cloning the Grafana repository, please refer to [GitHub's cloning a repository](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository) documentation.
+
+We do not recommend using Go's `go get` to download Grafana. Recent versions of Go have added behavior which isn't compatible with the way the Grafana repository is structured.
```
Comment (marcusolsson) [inline contribute/developer-guide.md:37]:
```suggestion
We don't recommend using Go's `go get` to download Grafana. Recent versions of Go have added behavior which isn't compatible with the way the Grafana repository is structured.
```

Comment (sakjur) [reply contribute/developer-guide.md:37]: I'm not a fan of contracting the emphasized part of a sentence. I now took a peek at our [documentation style guide](https://github.com/grafana/grafana/blob/34c2f440dbd701e31faa0949ddfc8921a8c6f460/contribute/style-guides/documentation-style-guide.md) and there isn't anything about this in particular, but we're using both don't and do not within that file.

I'll keep it as I wrote it for now, but I'm open to changing it given a rationale 🙂

Comment (aknuds1) [reply contribute/developer-guide.md:37]: I also prefer "do not" here, agree with @sakjur.

Comment (oddlittlebird) [reply contribute/developer-guide.md:37]: I would also change this to a warning.

> **Warning:** Do not use `go get` to download Grafana. Recent versions...

Comment (sakjur) [reply contribute/developer-guide.md:37]:
```suggestion
**Warning:** Do not use `go get` to download Grafana. Recent versions of Go have added behavior which isn't compatible with the way the Grafana repository is structured.
```

**Pull request merged**